tests/test_ml-notebook.py

```diff
@@ -23,10 +24,39 @@ def test_start():
         assert os.environ['PANGEO_ENV'] == 'ml-notebook'
 
 def test_jax_tf_together():
-    """ sometimes this impport fails due to sharing private symbols
-        complicated longer story, but it is better to
-        ensure they can coexist
     """
-    import tensorflow, jax
+    Sometimes this import fails due to sharing private symbols.
+    Complicated long story, but it is better to ensure they can coexist
+    """
+    import jax
+    import tensorflow
     assert int(tensorflow.__version__[0]) >= 2
     assert int(jax.__version__[0]) >= 0
+
+
+def test_jax_random_number_generator():
+    """
+    Ensure that initializing a random number generator on JaX works.
+
+    Regression test for checking that JaX and cuda-nvcc are installed and compatible on
+    GPU devices, see https://github.com/pangeo-data/pangeo-docker-images/issues/438.
+    """
+    import jax
+    import numpy as np
+    from jax import random
+
+    # Test running on CPU
+    with jax.default_device(jax.devices("cpu")[0]):
+        key = random.key(seed=42)
+        x = random.normal(key=key)
+        np.testing.assert_allclose(x, -0.18471177)
+
+    # Test running on GPU (need to run locally)
+    try:
+        gpu_device = jax.devices("gpu")[0]
+        with jax.default_device(gpu_device):
+            key = random.key(seed=24)
+            x = random.normal(key=key)
+            np.testing.assert_allclose(x, -1.168644)
+    except RuntimeError:  # Unknown backend: 'gpu' requested
+        logging.log(level=logging.INFO, msg="JAX was not tested on a GPU device")
```

Comment:
This doesn't work on GitHub Actions without a GPU, but I ran this locally using:

```
mamba create -n ml-notebook --file https://raw.githubusercontent.com/pangeo-data/pangeo-docker-images/2024.06.02/ml-notebook/conda-linux-64.lock
mamba activate ml-notebook
pytest --verbose tests/test_ml-notebook.py
```

The tests passed on my computer with an NVIDIA GPU, so should be ok I think.

Reply:
If there were Azure credits for Pangeo we could add GPU runners https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/managing-larger-runners !